features/FEATURE_BLE/source/generic/GenericGap.cpp

```diff
@@ -314,7 +314,9 @@ static bool is_whitelist_valid(const Gap::Whitelist_t& whitelist)
 
     for (size_t i = 0; i < whitelist.size; ++i) {
         const BLEProtocol::Address_t& address = whitelist.addresses[i];
-        if (address.type > BLEProtocol::AddressType::RANDOM_PRIVATE_NON_RESOLVABLE) {
```

Comment:
well that is a bit evil when it comes to enums when they get new items

Reply:
It is!
But it's not worst than a straight cast 😆 .